shared/actions/chat/attachment.js

```diff
@@ -118,55 +84,48 @@ function * onLoadAttachment ({payload: {conversationIDKey, messageID, loadPrevie
     identifyBehavior: RPCTypes.TlfKeysTLFIdentifyBehavior.chatGui,
   }
 
-  const channelConfig = singleFixedChannelConfig([
+  const channelConfig = Saga.singleFixedChannelConfig([
     'chat.1.chatUi.chatAttachmentDownloadStart',
     'chat.1.chatUi.chatAttachmentDownloadProgress',
     'chat.1.chatUi.chatAttachmentDownloadDone',
   ])
 
   const channelMap = ((yield call(ChatTypes.localDownloadFileAttachmentLocalRpcChannelMap, channelConfig, {param})): any)
 
-  const progressTask = yield effectOnChannelMap(c => safeTakeEvery(c, function * ({response}) {
+  const progressTask = yield Saga.effectOnChannelMap(c => Saga.safeTakeEvery(c, function * ({response}) {
     const {bytesComplete, bytesTotal} = response.param
-    const action: Constants.DownloadProgress = {
-      type: 'chat:downloadProgress',
-      payload: {conversationIDKey, messageID, isPreview: loadPreview || isHdPreview, bytesTotal, bytesComplete},
-    }
-    yield put(action)
+    yield put(Creators.downloadProgress(conversationIDKey, messageID, loadPreview || isHdPreview, bytesComplete, bytesTotal))
     response.result()
   }), channelMap, 'chat.1.chatUi.chatAttachmentDownloadProgress')
 
-  {
```

Comment:
This construct seems funky to me, having scope just cause we want to extract the variable out, just give it a name